quil-rs/src/instruction/gate.rs

```diff
@@ -0,0 +1,222 @@
+use std::fmt;
+
+use crate::{
+    expression::Expression,
+    validation::identifier::{validate_identifier, IdentifierValidationError},
+};
+
+use super::{format_qubits, get_expression_parameter_string, Qubit};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Gate {
+    pub name: String,
+    pub parameters: Vec<Expression>,
+    pub qubits: Vec<Qubit>,
+    pub modifiers: Vec<GateModifier>,
+}
```

Comment:
All of the instruction structs have public members. However, as we'll see in the `impl Gate` block below, it requires some knowledge to update them in a correct way. Should we strip away visibility into these members?

Reply:
Per offline discussion, we're letting this slide so we can press forward instead of adding more boiler plate.